### PR TITLE
Tharshana

### PR DESCRIPTION
**Pull request by stharshana**

### CODE REVIEW

**Comment by raj-mirabilis** [comment]
Please resolve the Conflicts in the Constants - EndpointURI, SubModuleDTO and ProjectSerImpl

File: defect-tracker-data-common/src/main/java/com/defect/tracker/data/dto/SubModuleDto.java [hunk 1]
```diff
@@ -2,18 +2,19 @@
 
 public class SubModuleDto {
 	
-	private Long subModId;
```
Comment (raj-mirabilis) [inline defect-tracker-data-common/src/main/java/com/defect/tracker/data/dto/SubModuleDto.java]: consider the ID - naming convention

File: defect-tracker-server/src/main/java/com/defect/tracker/util/Constants.java [hunk 2]
```diff
@@ -17,7 +17,8 @@
     
 	public static final String UpdateSubmodule= "SubModule updated successfully ";
 	public static final String SUBMODULE_DELETED ="SubModule Deleted";
-
```
Comment (raj-mirabilis) [inline defect-tracker-server/src/main/java/com/defect/tracker/util/Constants.java]: remove the unwanted spaces

File: defect-tracker-server/src/main/resources/application-dev.properties [hunk 3]
```diff
@@ -2,7 +2,7 @@ server.port=1724
 server.servlet.context-path=/defect-tracker
 spring.datasource.url=jdbc:mysql://localhost:3306/defect-tracker?allowPublicKeyRetrieval=true&useSSL=false
 spring.datasource.username=root
-spring.datasource.password=root
```
Comment (raj-mirabilis) [inline defect-tracker-server/src/main/resources/application-dev.properties:5]: please consider the application-dev properties - when adding and pushing

File: defect-tracker-data-common/src/main/java/com/defect/tracker/data/dto/SubModuleDto.java [hunk 4]
```diff
@@ -2,18 +2,19 @@
...
+	private Long id;
 	private String subModuleName;
 	private Long modId;
```
Comment (raj-mirabilis) [inline defect-tracker-data-common/src/main/java/com/defect/tracker/data/dto/SubModuleDto.java]: getter and setters

**Review by raj-mirabilis — CHANGES_REQUESTED** [review]
please the consider the corrections

File: defect-tracker-server/src/main/java/com/defect/tracker/controller/SubModuleController.java [hunk 5]
```diff
@@ -37,6 +37,15 @@
 	
 	@PutMapping(value= EndpointURI.UpdateSubModule)
 	public ResponseEntity<Object> updateSubModule(@RequestBody SubModuleDto subModuleDto){
+		/*
+		 * if(!subModuleService.existSubModule(subModuleDto.getId())) { return new
+		 * ResponseEntity<> (new
+		 * ValidationFailureResponse(ValidationConstance.SUBMODULE_NOT_EXIST,
```
Comment (ramz7) [inline defect-tracker-server/src/main/java/com/defect/tracker/controller/SubModuleController.java]: remove this commented code

File: defect-tracker-server/src/main/java/com/defect/tracker/services/DesignationServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+@Service
+public class DesignationServiceImpl implements DesignationService {
+	
+	@Autowired
+	DesignationRepository  designationRepository;
+	
+	@Override
+	public List<Designation> getAlldesignation() {
+		return designationRepository.findAll();
+
```
Comment (ramz7) [inline defect-tracker-server/src/main/java/com/defect/tracker/services/DesignationServiceImpl.java]: remove line break

File: defect-tracker-server/src/main/java/com/defect/tracker/services/ProjectSerImple.java [hunk 7]
```diff
@@ -39,6 +39,12 @@ public void deleteById(Long id) {
 		// TODO Auto-generated method stub
 		
 	}
+
+	@Override
+	public void updateProject(Project project) {
+		// TODO Auto-generated method stub
```
Comment (ramz7) [inline defect-tracker-server/src/main/java/com/defect/tracker/services/ProjectSerImple.java:46]: why this unimplemented method?

Comment (stharshana) [reply defect-tracker-server/src/main/java/com/defect/tracker/services/ProjectSerImple.java:46]: @Kajanthuka

File: defect-tracker-server/src/main/resources/application-dev.properties (hunk 3 above)
Comment (stharshana) [reply defect-tracker-server/src/main/resources/application-dev.properties:5]: ok I considered

**Comment by Kajanthuka** [comment]
This Service file might be deleted, In Action Have Another Service file  Exist => Name with ProjectServiceImpl,With All Overied Methods with Implementatation,

**Pull request merged**